cmd/crossplane/core/core.go

```diff
@@ -70,31 +75,41 @@ func (c *startCommand) Run(s *runtime.Scheme, log logging.Logger) error {
 	if err != nil {
 		return errors.Wrap(err, "Cannot get config")
 	}
-	log.Debug("Starting", "sync-period", c.Sync.String())
 
-	mgr, err := ctrl.NewManager(cfg, ctrl.Options{
+	mgr, err := ctrl.NewManager(ratelimiter.LimitRESTConfig(cfg, c.MaxReconcileRate), ctrl.Options{
 		Scheme:           s,
 		LeaderElection:   c.LeaderElection,
 		LeaderElectionID: "crossplane-leader-election-core",
```

Comment:
One strange issue I'm seeing with this PR is that somewhere between --max-reconcile-rate 50 and 100 we start to lose leader elections due to timing out while renewing the lease. I'm guessing there's some kind of starvation going on but, it's not immediately obvious to me where. We acquire the lease just fine, but fail to renew it. We're not hitting the API server particularly hard (~150rps) and the renewal fails whether I leave the deadline at the default 10 seconds, or bump it to 25.